stdlib/obj.mli

```diff
@@ -86,6 +86,7 @@ val int_tag : int
 val out_of_heap_tag : int
 val unaligned_tag : int   (* should never happen @since 3.11.0 *)
 
+type nonrec extension_constructor = extension_constructor
```

Comment:
What about creating an Extension_constructor sub-module, with a type `t` and three functions `of_val:'a -> t`, `name: t -> string` and `id: t -> int` (keeping the existing functions as deprecated aliases)?

Reply:
Good idea.  I've made that change.